### ISSUE TITLE
Cannot Build hellocardboard-ios - can't locate file for: -lProtobuf-C++

### ISSUE THREAD

**Issue by willbonde** [googlevr/cardboard#372]
### SPECIFIC ISSUE ENCOUNTERED
When attempting to build the hellocardboard-ios sample code, I've run into the following issue after following the directions at https://developers.google.com/cardboard/develop/ios/quickstart:

```
Create static library GfxPluginCardboard.a (x86_64)

Libtool /Users/will/projects/code/thirdparty/cardboard/sdk/build/Debug-iphonesimulator/GfxPluginCardboard.a normal (in target 'sdk' from project 'sdk')
    cd /Users/will/projects/code/thirdparty/cardboard/sdk
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool -static -arch_only x86_64 -D -syslibroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.0.sdk -L/Users/will/projects/code/thirdparty/cardboard/sdk/build/Debug-iphonesimulator -L/Users/will/projects/code/thirdparty/cardboard/sdk/build/Debug-iphonesimulator/Protobuf-C++ -filelist /Users/will/projects/code/thirdparty/cardboard/sdk/build/sdk.build/Debug-iphonesimulator/sdk.build/Objects-normal/x86_64/GfxPluginCardboard.LinkFileList -lProtobuf-C++ -framework Metal -framework MetalKit -dependency_info /Users/will/projects/code/thirdparty/cardboard/sdk/build/sdk.build/Debug-iphonesimulator/sdk.build/Objects-normal/x86_64/GfxPluginCardboard_libtool_dependency_info.dat -o /Users/will/projects/code/thirdparty/cardboard/sdk/build/Debug-iphonesimulator/GfxPluginCardboard.a

error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: can't locate file for: -lProtobuf-C++
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: -lProtobuf-C++ is not an object file (not allowed in a library)
Command Libtool failed with a nonzero exit code
```

I'm assuming this is a user error, but I would highly appreciate some guidance as I did follow the 6 steps in the guide. I appreciate the Cardboard team!

### SMARTPHONE (please complete the following information):
- Device: iPhone 14 Pro (sim and physical device)
- OS
  - [ ] Android
  - [x] iOS
- OS version: 16

### STEPS TO REPRODUCE THE ISSUE
1. Follow instructions on https://developers.google.com/cardboard/develop/ios/quickstart

### EXPECTED BEHAVIOR
Project would build

### VERSIONS USED
**What version of Google Cardboard are you using?**
Cardboard SDK 1.17.0 (master as of 9/24/2022)

- Select in which graphics API you can reproduce the bug.
   - [ ] OpenGL ES2
   - [ ] OpenGL ES3
   - [x] Metal

Note: I can't build it so likely moot

### LINK TO A REPOSITORY WITH CODE TO REPRODUCE THE BUG
Using sample project in this repository, no different code

### SCREENSHOTS
n/a

### ADDITIONAL CONTEXT
-XCode Version 14.0 (14A309)
-macOS 12.6 with an Intel processor
-CocoaPods 1.11.3

**Comment by arilow** [comment]
Thanks for reaching out to us @michal-repo! Unfortunately we couldn’t reproduce the issue using Windows 10 and Android Studio Dolphin 2021.3.1.

Nevertheless, based on the error log, apparently either the SDK wasn’t built or its binary and header file weren’t copied into the [libraries/ folder](https://github.com/googlevr/cardboard/blob/master/hellocardboard-android/build.gradle#L61). Have you followed step 3 of [Download and build the demo app](https://developers.google.com/cardboard/develop/c/quickstart#download_and_build_the_demo_app) from the quickstart?
Once it is done, if you go to `{YOUR_PROJECT_PATH}/cardboard/sdk/build/outputs/aar/` you should find 2 files:
- sdk-debug.aar
- sdk-release.aar

And then you can try to build the Hello Cardboard sample app again.

In case it keeps failing, please check that a folder called “libraries” has been created with the following content within [hellocardboard-android](https://github.com/googlevr/cardboard/tree/master/hellocardboard-android):

<img width="251" alt="Screen Shot 2022-09-28 at 11 29 30" src="https://user-images.githubusercontent.com/43258762/192806710-b61440c6-63a3-4b5f-a331-e3ef7600ca04.png">

In case this does not solve your issue, please reach out to us again.

Thanks!

**Comment by willbonde** [comment]
Thanks for replying @arilow - was the response meant for this issue? The build problem was encountered building the iOS version, not Android and @michael-repo wasn't on the thread. I was thinking perhaps the response was meant for another thread :)

**Comment by arilow** [comment]
Hi @willbonde. Yes, the response was meant for another issue, sorry and thanks for letting me know!

About this issue: I have tried to reproduce the error using Xcode 13.2 and I couldn't. I will try to reproduce it using Xcode 14 and let you know the results as soon as possible.

Thanks!

**Comment by willbonde** [comment]
@arilow no problem - I do that all the time lol. I look forward to hearing more, let me know if I can help or provide more info.

**Comment by arilow** [comment]
Hi again @willbond! I couldn’t reproduce this issue with Xcode 14 either.

Would you be so kind as to zip your Xcode project and share it with us? That would allow us to try to build it on our side and find the issue.

Thanks!

**Comment by willbonde** [comment]
@arilow  Sure thing - will get it to you this afternoon!

**Comment by willbonde** [comment]
@arilow Here it is! I cleaned the build folder before zipping it to reduce file size... let me know if you want that. I did save it.

[cardboard.zip](https://github.com/googlevr/cardboard/files/9676253/cardboard.zip)

For reference, here's my process:


## Command Line:


```
[will@Wills-MacBook-Pro tmp]# git clone https://github.com/googlevr/cardboard.git
Cloning into 'cardboard'...
remote: Enumerating objects: 1405, done.
remote: Counting objects: 100% (451/451), done.
remote: Compressing objects: 100% (203/203), done.
remote: Total 1405 (delta 293), reused 297 (delta 235), pack-reused 954
Receiving objects: 100% (1405/1405), 12.25 MiB | 8.73 MiB/s, done.
Resolving deltas: 100% (694/694), done.
[will@Wills-MacBook-Pro tmp]# cd cardboard/
[will@Wills-MacBook-Pro cardboard]# pod install
Analyzing dependencies
Downloading dependencies
Installing Protobuf-C++ (3.18.0)
Generating Pods project
Integrating client project
Pod installation complete! There is 1 dependency from the Podfile and 1 total pod installed.

[!] The Podfile contains framework or static library targets (sdk), for which the Podfile does not contain host targets (targets which embed the framework).
If this project is for doing framework development, you can ignore this message. Otherwise, add a target to the Podfile that embeds these frameworks to make this message go away (e.g. a test target).

[!] Your project does not explicitly specify the CocoaPods master specs repo. Since CDN is now used as the default, you may safely remove it from your repos directory via pod repo remove master. To suppress this warning please add warn_for_unused_master_specs_repo => false to your Podfile.
[will@Wills-MacBook-Pro cardboard]# open Cardboard.xcworkspace
```


## In Xcode:

- Changed bundle id to com.williambonde.HelloCardboard
- Went to sdk > Build Phases > Link Binary With Libraries
  - Removed libPods-sdk.a
  - Added libProtobuf-C++.a. Click "Add Anyway"
- Cleaned build folder (unneeded, force of habit, just documenting)
- Selected "HelloCardboard" target, iPhone 14 Pro
- Hit Run

Result:

- sdk: "error build: Command Libtool failed with a nonzero exit code", Details:

```
Libtool /Users/will/tmp/cardboard/sdk/build/Debug-iphonesimulator/GfxPluginCardboard.a normal (in target 'sdk' from project 'sdk')
    cd /Users/will/tmp/cardboard/sdk
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool -static -arch_only x86_64 -D -syslibroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.0.sdk -L/Users/will/tmp/cardboard/sdk/build/Debug-iphonesimulator -L/Users/will/tmp/cardboard/sdk/build/Debug-iphonesimulator/Protobuf-C++ -filelist /Users/will/tmp/cardboard/sdk/build/sdk.build/Debug-iphonesimulator/sdk.build/Objects-normal/x86_64/GfxPluginCardboard.LinkFileList -lProtobuf-C++ -framework Metal -framework MetalKit -dependency_info /Users/will/tmp/cardboard/sdk/build/sdk.build/Debug-iphonesimulator/sdk.build/Objects-normal/x86_64/GfxPluginCardboard_libtool_dependency_info.dat -o /Users/will/tmp/cardboard/sdk/build/Debug-iphonesimulator/GfxPluginCardboard.a

error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: can't locate file for: -lProtobuf-C++
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: file: -lProtobuf-C++ is not an object file (not allowed in a library)
Command Libtool failed with a nonzero exit code

```

**Comment by willbonde** [comment]
Hey @arilow, I was able to tweak the project and get it to build. I will type up some notes later. Just wanted to give you heads up so you don't waste any time.

I am perplexed why I had to change some paths - that'll be in my notes - I would be curious your thoughts on that. Stand by, and thank you! :)

**Comment by arilow** [comment]
Closing as this problem seems to be fixed.

@willbonde, we would really appreciate it if you could share your notes regarding this.

Thanks!

**Comment by willbonde** [comment]
@arilow My goodness, I'm so sorry I forgot to follow up.

For me, the only thing I had to change was the Library Search path is the `sdk` Target. It was `$(inherited)`, which resolved to `build/Debug-iphoneos/Protobuf-C++`. 

I changed it to `../build/Debug-iphoneos/Protobuf-C++`, and then the build succeeds.
